src/lfortran/semantics/ast_body_visitor.cpp

```diff
@@ -2108,6 +2122,11 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
             increment = nullptr;
         }
 
+        if (var && var->type == ASR::exprType::Var) {
```

Comment:
I think it must be a variable, doesn't it?
```suggestion
        if (var) {
```
That way we get an assert failure if it isn't, which signals a bug in our compiler, instead of ignoring it.

Reply:
> I think it must be a variable, doesn't it?

I guess yes, to assure it I kept the condition for `var->type`.